Validate that searchable copies matches redundancy

bjorncs opened the pull request:
Verify that the number of searchable copies is the same as effective
redundancy. Rewrite GlobalDistributionValidatorTest to not use Mockito.
Add end-to-end to verify that the GlobalDistributionValidator is wired
into the overall model validation.

bjorncs committed:
Validate that searchable copies matches redundancy

Verify that the number of searchable copies is the same as effective
redundancy. Rewrite GlobalDistributionValidatorTest to not use Mockito.
Add end-to-end to verify that the GlobalDistributionValidator is wired
into the overall model validation.

geirst reviewed (changes requested):
Overall looks good, please look at my nitpick comments

bjorncs committed:
Change year in copyright header from 2016 to 2017

bjorncs committed:
Don't add statistics property as trace level 0

Fix bug where statistics are always visible in trace output,
independent of trace level. Also remove duplicate visitor statistics in trace.
